Library/Homebrew/brew.sh

```diff
@@ -305,7 +318,12 @@ auto-update() {
       timer_pid=$!
     fi
 
-    brew update --auto-update
+    if [[ -n "${HOMEBREW_AUTO_UPDATE_CORE_CLONES}" ]]
+    then
+      brew update --auto-update --core-clones
```

Comment:
```suggestion
      brew update
```

Wondering if this might be a more simple change here as this already will update both of these and avoids adding Yet Another Flag to `brew update` and, if I've read things correctly, will remove the need for any changes to `update.sh` entirely.

Reply:
> will remove the need for any changes to `update.sh` entirely.

Well, there will still be the need to modify the conditionals that skip updating core and cask.

Reply:
> Fine with this if you feel there's no use case for a manual `brew update --core-clones`.

Yeh, I think there's not. Can just manually run `brew update`.



> Well, there will still be the need to modify the conditionals that skip updating core and cask (the lines modified to check for `HOMEBREW_UPDATE_CORE_CLONES`)

Gotcha, fewer changes at least then 👍🏻

Reply:
> Can just manually run `brew update`.

You mean `git pull`? `brew update` won't touch your taps by default.

Reply:
Maybe's it's `HOMEBREW_DEVELOPER` only but should be `devcmdrun` only?

Reply:
Possibly. Though not everyone with `devcmdrun` is interested in core/cask taps (e.g. brew contributors, 3rd party tap maintainers) and we save a ton of time skipping them.

Ideally there would be a mode for core contributors I guess?

Reply:
> Possibly. Though not everyone with `devcmdrun` is interested in core/cask taps (e.g. brew contributors, 3rd party tap maintainers) and we save a ton of time skipping them.

Yeh, my thinking is that this is just for people running `brew update` explicitly or a dev-cmd that's doing an auto-update.

> Or maybe we just consider `devcmdrun` users that have the tap updated since 4.0.0's release date? Kinda funky but would filter out those who only have a tap because of having Homebrew installed for a while.

Could combine this sort of tweak with https://github.com/Homebrew/brew/pull/14592 landing as then (or around then) we should start nagging people to consider untapping homebrew/core or homebrew/cask if they don't show any evidence of needing them.

Reply:
> Yeh, my thinking is that this is just for people running `brew update` explicitly or a dev-cmd that's doing an auto-update.

dev-cmd auto-update is basically this PR yeah, with the only difference being `brew release`.

I do agree that we could do something better for an explicit `brew update`. It might be worth creating an issue about it and deferring it to a separate PR, since I reckon that behaviour would be extra functionality rather than replacing any auto-update behaviour - we _always_ want to update taps for bump commands but for explicit `brew update` we probably want to be smarter and acknowledge taps that are several months out of date which could take several minutes to update.

Reply:
> I do agree that we should do something better for an explicit `brew update`. There's some ideas here so it might be worth creating an issue about it and deferring it to a separate PR, since I reckon that behaviour would be extra functionality rather than replacing any auto-update behaviour, since we always want to update taps for bump commands (unless we're bumping a third party tap technically) but for explicit `brew update` we probably want to be smarter and acknowledge taps that are several months out of date which could take several minutes to update.

Fine with me 👍🏻. As long as we're not adding the new flag, I think the rest is fine to punt to another PR.